src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -925,7 +928,7 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
         case other =>
           fatal(
             s"""method 'annotate_variants_table' expects a key table keyed by one of the following:
-               |  [ Variant ]
+               |  [ $vSignature ]
```

Comment:
Why janky?  That's literally what the code above says:

 >         case Array(`vSignature`) =>

I agree, in the long term we could reasonably support conversions from the function registry in matching up keys on joins.